src/app/components/SwapSubmitModal/index.jsx

```diff
@@ -1,52 +1,134 @@
+import { push } from 'react-router-redux'
+import toastr from 'Utilities/toastrWrapper'
+import log from 'Utilities/log'
 
-import SwapSubmitModalView from './view'
-
-class SwapSubmitModal extends Component {
-  constructor (props) {
-    super(props)
-    this.handleCancel = this.handleCancel.bind(this)
-    this.handleSignTxs = this.handleSignTxs.bind(this)
-    this.handleSendTxs = this.handleSendTxs.bind(this)
-  }
-
-  handleCancel () {
-    const { swundle, toggle, removeSwundle } = this.props
-    closeTrezorWindow()
-    toggle()
-    removeSwundle(swundle)
-  }
+const SwapSubmitModal = ({
+  isOpen, swundle, headerText, continueText, continueDisabled, continueLoading,
+  errorMessage, handleContinue, handleCancel, currentSwap, secondsUntilPriceExpiry, showTimer, shouldShowTimer
+}) => (
+  <div>
+    <Modal size='lg' backdrop='static' isOpen={isOpen} toggle={handleCancel}>
+      <ModalHeader className='text-primary' toggle={handleCancel}>
+        {headerText}
+      </ModalHeader>
+      <ModalBody className='modal-text'>
+        {errorMessage && (
+          <Alert color='danger'>{errorMessage}</Alert>
+        )}
+        <p>
+          The following swaps will take place to save the changes you made to your portfolio. Please review them and click {`"${continueText}"`} to proceed.
+        </p>
+        <div className='my-3'>
+          <Row className='gutter-2'>
+            {swundle.swaps.map((swap) => {
+              const { id, tx, status: { code, detailsCode, labelClass, label } } = swap
+              let statusText
+              if (detailsCode === 'signed') {
+                statusText = (<span className='text-success'>Signed</span>)
+              } else if (detailsCode === 'signing_unsupported') {
+                statusText = (<span className='text-success'>Ready</span>)
+              } else if (detailsCode === 'signing') {
+                statusText = (<span className='text-warning blink'>Awaiting signature</span>)
+              } else if (detailsCode.includes('error')) {
+                statusText = (<span className='text-danger'>Failed</span>)
+              } else if (detailsCode === 'sending') {
+                statusText = (<span className='text-primary'>Sending</span>)
+              } else if ((tx && tx.sent) || code === 'failed') {
+                statusText = (<span className={labelClass}>{label}</span>)
+              } else if (detailsCode !== 'unsigned') {
+                statusText = (<Spinner size='sm' inline/>)
+              }
+              return (
+                <Col xs='12' key={id}>
+                  <SwapStatusCard swap={swap} statusText={statusText} />
+                </Col>
+              )
+            })}
+          </Row>
+        </div>
+        <p>Total network fee: {swundle.totalTxFee
+          ? display.fiat(swundle.totalTxFee)
+          : <Spinner inline size='sm'/>}
+        </p>
+        {(showTimer && secondsUntilPriceExpiry > 0)
```

Comment:
i left this for now, need a bit of clarification!

Reply:
Oh I see what's going on now. I was confused by the `showTimer` naming. Can you try this instead:
```javascript
withStateHandlers(
  { timerExpired: false },
  { handleTimerEnd: () => () => ({ timerExpired: true }) },
)
```
And I was wrong about your condition, there's only one simplification:
```javascript
{(secondsUntilPriceExpiry > 0 && !timerExpired)
  ? (...<Timer onTimerEnd={handleTimerEnd} .../>)
  : (timerExpired && (...Fixed rate is no longer guaranteed...))}
```